### ISSUE TITLE
An unproductive spec

### ISSUE THREAD

**Issue by christianbean** [PermutaTriangle/Tilings#206]
A recursion depth bug was found because of an infinite loop while enumerating 01234_01243_10234_10243. If you look at the tree https://combopal.ru.is/draw/garpur?run_id=5efa9e6ae610a9638ca9c083, and look at the bottom left branch, there is the chain 17 -> 21 -> 17 , each individual step is fine, but together we are doing a complicated 17 = 17 step - so this is a tautology. You can see what is going on is that the assumption that we split is precisely the assumption we add back in:

```
Splitting the assumptions
T1                                                T2
Assumption 0:                                     Assumption 0:                                
can count points in cell (0, 1)                   can count points in cell (0, 1)              
Assumption 1:                                     Assumption 1:                                
can count points in cells (0, 1), (1, 0)          can count points in cell (1, 0)              
Assumption 2:                                     
can count points in cell (1, 0)       

Adding the assumptions
T2                                                T1
Assumption 0:                                     Assumption 0:                                
can count points in cell (0, 1)                   can count points in cell (0, 1)              
Assumption 1:                                     Assumption 1:                                
can count points in cell (1, 0)                   can count points in cells (0, 1), (1, 0)     
                                                  Assumption 2:                                
                                                  can count points in cell (1, 0)  
```

Two potential fixes:
1) We shouldn't add an assumption which is a disjoint union of other assumptions. (This might not fix everything)
2) the splitting strategy is currently used as a "pre strategy" to get ready for factoring, we could do it all it once, removing the need for splitting strategy altogether.

**Comment by ulfarsson** [comment]
Solution nr. 2 feels like a better long-term solution. If you like having two steps in the spec, you could perhaps move the assumption work into the factor strategy like you suggest, but make the strategy output two rules when it applies.

**Comment by jaypantone** [comment]
Another example: https://api.combopal.ru.is/garpur_run/5efe275ab754ee7e68833dd4

**Comment by christianbean** [comment]
fixed by #216
